pkg/controllers/util/federatedclient/podinformer_test.go

```diff
@@ -0,0 +1,236 @@
+package federatedclient
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	kubeinformer "k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
+)
+
+func setupFakeClient(numPods int) (pods []runtime.Object, client kubernetes.Interface) {
+	pods = make([]runtime.Object, numPods)
+	for i := 1; i <= numPods; i++ {
+		pods[i-1] = newPod(i)
+	}
+	return pods, fake.NewSimpleClientset(pods...)
+}
+
+func newPod(id int) *corev1.Pod {
+	return &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:   fmt.Sprintf("test-pod-%d", id),
+			Labels: map[string]string{"foo": "bar"},
+		},
+		Spec: corev1.PodSpec{
+			Overhead: corev1.ResourceList{
+				corev1.ResourceCPU: resource.MustParse("1m"),
+			},
+			InitContainers: []corev1.Container{
+				{
+					Name: "initcontainer",
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("1m"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("1m"),
+						},
+					},
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name: "container",
+					Resources: corev1.ResourceRequirements{
+						Limits: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("10m"),
+						},
+						Requests: corev1.ResourceList{
+							corev1.ResourceCPU: resource.MustParse("10m"),
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func Test_federatedClientFactory_podInformer(t *testing.T) {
```

Comment:
It looks like the pod informer can be tested in isolation from `federatedClientFactory`. If possible, I think it's preferable to coupling the two.